DeviceSDK/src/main/java/org/astarteplatform/devicesdk/AstarteDevice.java

```diff
@@ -40,6 +35,42 @@ public boolean hasInterface(String interfaceName) {
     return mAstarteInterfaces.containsKey(interfaceName);
   }
 
+  /**
+   * Method that adds an interface dynamically to the introspection
+   *
+   * @param astarteInterfaceObject the JSONObject representation of the interface
+   * @throws AstarteInvalidInterfaceException when the JSON Object does not represent an interface
+   *     correctly
+   * @throws AstarteInterfaceAlreadyPresentException when an interface with the same name, major and
+   *     minor is already present
+   */
+  public void addInterface(JSONObject astarteInterfaceObject)
```

Comment:
What happens if you call this with an interface with the same name and major but a different minor?
The expected behaviour I'm thinking about is:
- Same minor: exception is raised (as it is right now)
- New minor < old minor: exceptions is raised (can't downgrade an interface at runtime)
- New minor > old minor: the new interface replaces the old one

Please confirm if this is the actual behaviour